lib/src/command/outdated.dart

```diff
@@ -185,14 +224,42 @@ class OutdatedCommand extends PubCommand {
     }
   }
 
+  /// Get the latest version of [ref] given its [PackageRef].
+  ///
+  /// Returns `null`, if unable to find the package.
+  Future<PackageId> _getLatest(PackageRef ref) async {
+    if (ref == null) {
+      return null;
+    }
+    try {
+      final available = await cache.source(ref.source).getVersions(ref);
+      if (available.isEmpty) {
+        return null;
+      }
+      available.sort(argResults['pre-releases']
+          ? (x, y) => x.version.compareTo(y.version)
+          : (x, y) => Version.prioritize(x.version, y.version));
+      return available.last;
+    } on PackageNotFoundException {
```

Comment:
When do we get this? Can we avoid this, I think it means we are asking about a bad package.

Reply:
> Can we avoid this, I think it means we are asking about a bad package.

Maybe we make an `assert(false)` here and write, this should not be possible.
But in production I would prefer to handle the case..

I think this can happen with `--no-dependency-override`, if you wrote a package that doesn't exist in the `dependencies:` section of your `pubspec.yaml` -- and then proceeded to override it.

Actually, you're probably right we should not handle this.